server/plugin/webhook.go

```diff
@@ -1382,3 +1392,90 @@ func (p *Plugin) postReleaseEvent(event *github.ReleaseEvent) {
 		}
 	}
 }
+
+func (p *Plugin) postDiscussionEvent(event *github.DiscussionEvent) {
+	repo := event.GetRepo()
+
+	subs := p.GetSubscribedChannelsForRepository(repo)
+	if len(subs) == 0 {
+		return
+	}
+
+	newDiscussionMessage, err := renderTemplate("newDiscussion", event.GetDiscussion())
+	if err != nil {
+		p.client.Log.Warn("Failed to render template", "error", err.Error())
+		return
+	}
+
+	for _, sub := range subs {
+		if !sub.Discussions() {
+			continue
+		}
+
+		if p.excludeConfigOrgMember(event.GetSender(), sub) {
+			continue
+		}
+
+		post := &model.Post{
+			UserId:  p.BotUserID,
+			Type:    "custom_git_discussion",
+			Message: newDiscussionMessage,
+		}
+		repoName := strings.ToLower(repo.GetFullName())
+		discussionNumber := event.GetDiscussion().GetNumber()
+
+		post.AddProp(postPropGithubRepo, repoName)
+		post.AddProp(postPropGithubObjectID, discussionNumber)
+		post.AddProp(postPropGithubObjectType, "discussion")
+		post.ChannelId = sub.ChannelID
+		if err = p.client.Post.CreatePost(post); err != nil {
+			p.client.Log.Warn("Error webhook post", "post", post, "error", err.Error())
+		}
+	}
+}
+
+func (p *Plugin) postDiscussionCommentEvent(event *github.DiscussionCommentEvent) {
+	repo := event.GetRepo()
+
+	subs := p.GetSubscribedChannelsForRepository(repo)
+	if len(subs) == 0 {
+		return
+	}
+
+	if event.GetAction() != actionCreated {
+		return
+	}
+
+	newDiscussionCommentMessage, err := renderTemplate("newDiscussionComment", event)
+	if err != nil {
+		p.client.Log.Warn("Failed to render template", "error", err.Error())
+		return
+	}
+	for _, sub := range subs {
+		if !sub.DiscussionComments() {
+			continue
+		}
+
+		if p.excludeConfigOrgMember(event.GetSender(), sub) {
+			continue
+		}
+
+		post := p.makeBotPost("", "custom_git_dis_comment")
+
+		repoName := strings.ToLower(repo.GetFullName())
+		commentID := event.GetComment().GetID()
+
+		post.AddProp(postPropGithubRepo, repoName)
+		post.AddProp(postPropGithubObjectID, commentID)
+		post.AddProp(postPropGithubObjectType, githubObjectTypeDiscussionComment)
+
+		if event.GetAction() == actionCreated {
+			post.Message = newDiscussionCommentMessage
+		}
```

Comment:
We can remove this and directly pass the message int `createBotPost` func. We have already added the conditional for `actionCreated` on line 1445